don't check 0 elements for cat symbolic diff

ngimel opened the pull request:
Summary: Fixes symbolic script grad formula for cat to correctly handle empty tensors

Test Plan: Existing tests

Differential Revision: D31208364

eellison commented on torch/csrc/jit/runtime/symbolic_script.cpp:
```diff
@@ -612,7 +612,7 @@ const std::vector<std::string> functions = {
             split_sizes = [0] * size
             for i in range(size):
-                if tensors[i].numel() > 0:
+                if tensors[i].numel() > 0 or len(tensors[i].size()) != 1:
```
i guess this is the equivalent logic : https://codebrowser.bddppq.com/pytorch/pytorch/torch/csrc/autograd/FunctionsManual.cpp.html#602

expressed in python as tensors[i].size() != [0]. should we do that here as well ?

eellison commented on torch/csrc/jit/runtime/symbolic_script.cpp:
```diff
@@ -612,7 +612,7 @@ const std::vector<std::string> functions = {
             split_sizes = [0] * size
             for i in range(size):
-                if tensors[i].numel() > 0:
+                if tensors[i].numel() > 0 or len(tensors[i].size()) != 1:
```
nit: `len(tensors[i].size()) != 1` -> `tensors[i].dim() != 1`

ngimel replied:
heh, I tried `.ndim` and `.ndimension()`, and neither compiled with script, I guess `.dim()`'s the ticket. But you are right above, these 2 conditions are equivalent to `tensors[i].size() != [0]`

eellison replied:
`.size() != [0]` should work, .size() returns a list of ints in TS..

ngimel replied:
Ah interesting, it doesn't work in eager
```
In [8]: a.size() == [0]
Out[8]: False

In [9]: a.size() == torch.Size([0])
Out[9]: True
```

eellison reviewed (approved):
lgtm!! thanks!!

Commit:
don't check 0 elements for cat symbolic diff (#65751)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/65751

Fixes symbolic script grad formula for cat to correctly handle empty tensors

Test Plan: Existing tests

Reviewed By: eellison

Differential Revision: D31208364

fbshipit-source-id: eed718504c92c5bf90302dd11832b81dbc3f958d
